src/Typechecker/Typechecker.hs

```diff
@@ -26,6 +26,16 @@ import qualified Data.Set as S
 -- | return Nothing or the first element of a list which doen't satisfy a predicate
 all' p xs = foldl (\none x -> if p x then none else Just x) Nothing xs
 
+
+-- | Return a list of the covered positions for a board equation
+getCovered :: (Int,Int) -> BoardEq SourcePos -> [(Int,Int)]
+getCovered (mx,my) (PosDef _ xp yp _) = case (xp,yp) of
+                                  (ForAll _,ForAll _) -> fmap (\x -> (x `mod` mx, x `div` my)) [1..(mx*my)]
```

Comment:
The first case will generate the incorrect indices (e.g. `(mx, my) = (1,1)`). Use `[(x,y) | x <- [1..mx], y <- [1..my]]` instead. Set builder notation is cleaner for the next two cases as well. 

However, we [already create a list of indices when evaluating the board](https://github.com/The-Code-In-Sheep-s-Clothing/Spiel-Lang/blob/c6ce34d5bd7b0515dcc96c8dfb280532b7b677b6/src/Runtime/Eval.hs#L58), so now we are duplicating work. Perhaps the cleanest solution would be to add an argument for the positions to `BVal` and generate it when parsing? This would also make the type checker implementation closer to the mathematical specification that I am working on.

The `[BoardEq a]` then becomes redundant for evaluation, but it would still be useful for reporting errors.

Reply:
Can't think of how we would do it at parse time, I imagine we would still need to do it within the TC environment.

As for the original, yeah a list comprehension would be a better way of setting that up, good catch on the bad logic I had in place. That would work well for those other cases as well.